.github/workflows/ci.yml

```diff
@@ -0,0 +1,174 @@
+        description: 'If true, this will skip the release job'
+        type: boolean
+        default: false
+      release_branches:
+        description: 'Required if the branch that should process releases is in a non-standard format (e.g. main or MOODLE_XX_STABLE)'
+        type: string
+      moodle_branches:
+        description: 'Specify the MOODLE_XX_STABLE branch you want to test against'
+        type: string
+
+jobs:
+  prepare_matrix:
+    name: prepare test matrix
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+    runs-on: 'ubuntu-latest'
+    outputs:
+      matrix: ${{ steps.parse-version.outputs.matrix }}
+      component: ${{ steps.parse-version.outputs.component }}
+      highest_moodle_branch: ${{ steps.parse-version.outputs.highest_moodle_branch }}
+      release_required: ${{ (
+          contains(github.event_name, 'push') &&
+          steps.check-version.outputs.any_changed == 'true' &&
+          steps.check-branch.outputs.publishable == 'true'
+        ) }}
+    steps:
+      - name: Check if currently on a publishable branch
+        id: check-branch
+        run: |
+          publishable=${{ github.event.repository.fork == false &&
+          inputs.disable_release == false  && (
+            (startsWith(github.ref, 'refs/heads/MOODLE_') &&
+            endsWith(github.ref, '_STABLE')) ||
+            github.ref == 'refs/heads/main' || (
+              inputs.release_branches != '' &&
+              endsWith(github.ref, inputs.release_branches)
+            )
+          ) }}
+
+          echo '::set-output name=publishable::$publishable'
+      - name: Check out CI code
+        uses: actions/checkout@v2
+        with:
+          path: ci
+          repository: catalyst/catalyst-moodle-workflows
+          ref: v2
+          token: ${{ github.token }}
+      - name: Check out plugin code
+        uses: actions/checkout@v2
+        with:
+          # Needed for 'changed-files' actions (alternatively could be a fixed
+          # large number but may cause issues if limited).
+          fetch-depth: ${{ (contains(github.event_name, 'push') && steps.check-branch.outputs.publishable == 'true') && 0 || 1 }}
+          path: plugin
+      - name: Check if release is required (version.php changes)
+        if: contains(github.event_name, 'push') && steps.check-branch.outputs.publishable == 'true'
+        uses: tj-actions/changed-files@v18.7
+        id: check-version
+        with:
+          path: plugin
+          files: version.php
+          since_last_remote_commit: "true"
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.1'
+          coverage: none
+          # extensions: yaml # FYI: makes the workflow very slow
+      - name: Determine test requirements and plugin info
+        id: parse-version
+        run: |
+          chmod +x "${GITHUB_WORKSPACE}/ci/.github/actions/parse-version/script.sh"
```

Comment:
Not for this exercise yes you're right. It's no longer required. Originally it was going to do some things in bash land but functionally speaking, none of it is required. Not even the outputting of the php version. It can all be done inline here. 

So follow up question, do we want to do the bash related bits inline here if there were more things to check in the future? For instance if we decided we wanted to check the previous git version a different way without using the github actions (tj-actions/changed-files@v18.7)